sklearn/tests/test_multioutput.py

```diff
@@ -417,6 +418,8 @@ def test_multi_output_exceptions():
     moc.fit(X, y)
     with pytest.raises(ValueError):
         moc.score(X, y_new)
+    with pytest.raises(AttributeError, match=msg):
```

Comment:
I think that we can slightly rework the test here to isolate the `NoFittedError` and the attribute delegation in 2 separate tests and let the focus the remainder of the test only on the difference of output between fit and predict. Here comes my proposal:

```python
@pytest.mark.parametrize("response_method", ["predict_proba", "predict"])
def test_multi_output_notfittederror(response_method):
    """Check that we raise the proper error when the estimator is not fit."""
    clf = MultiOutputClassifier(LogisticRegression())
    with pytest.raises(NotFittedError):
        getattr(clf, response_method)(X)


def test_multi_output_delegate_predict_proba():
    """Check the behavior for the delegation of predict_proba to the underlying
    estimator."""

    # A base estimator with `predict_proba`should expose the method even
    # before fit
    clf = MultiOutputClassifier(LogisticRegression())
    assert hasattr(clf, "predict_proba")
    clf.fit(X, y)
    assert hasattr(clf, "predict_proba")

    # A bas estimator without `predict_proba` should raise an AttributeError
    clf = MultiOutputClassifier(LinearSVC())
    assert not hasattr(clf, "predict_proba")
    with pytest.raises(AttributeError):
        clf.predict_proba(X)
    clf.fit(X, y)
    assert not hasattr(clf, "predict_proba")
    with pytest.raises(AttributeError):
        clf.predict_proba(X)
```